src/neural/cuda/network_cudnn.cc

```diff
@@ -310,6 +310,26 @@ class CudnnNetwork : public Network {
     if (!options.IsDefault<bool>("custom_winograd"))
       use_custom_winograd_ = options.Get<bool>("custom_winograd");
 
+    // Warn if the memory required for storing transformed weights is
+    // going to exceed 60% of total video memory, force custom_winograd off
+    // if it's going to exceed 80% of memory.
+    size_t residual_weight_size =
+        3 * 3 * kNumFilters * kNumFilters * numBlocks_ * sizeof(DataType);
+    size_t transformed_residual_weight_size = residual_weight_size * 4;
```

Comment:
Shouldn't this logic be conditional on use_custom_wingrad not already being false?

Reply:
I would split this in two, moving the logic for `use_custom_winograd_ = false;` before line 310 (`if (!options.IsDefault<bool>("custom_winograd"))`) and keep only the warning here (conditional as Tilps says).

Reply:
Yes. I wanted to have some warning for people with very less memory even with custom_winograd truend off but we probably don't want people to get confused by useless messages. Updated the condition so that we print only useful messages.